.github/workflows/ubuntu-java.yml

```diff
@@ -79,19 +82,21 @@ jobs:
 
       - name: make
         run: |
-          cd _build
-          make --jobs=$(($(nproc)+1))
-
-      # - name: check
-      #   run: |
-      #     cd _build
-      #     make check TESTSUITEFLAGS="--jobs=$(($(nproc)+1))"
+          make -C _build --jobs=$(($(nproc)+1))
 
-      # note: distcheck also creates the dist tarball
       - name: distcheck
         run: |
-          cd _build
-          make --jobs=$(($(nproc)+1)) distcheck TESTSUITEFLAGS="--jobs=$(($(nproc)+1))"
+          make -C _build distcheck                                    \
+                         TESTSUITEFLAGS="--jobs=$(($(nproc)+1))"      \
+                         DISTCHECK_CONFIGURE_FLAGS="--without-indexed \
+                                                    --without-json    \
+                                                    --without-xml2    \
+                                                    --with-java" ||   \
+          make -C _build check                                        \
+                         TESTSUITEFLAGS="--recheck --verbose"
```

Comment:
this won't work, the `make checkl` within distcheck are created in a subfolder, so you'd have to rerun it there

```sh
          make -C _build _build/gnucobol-4.0/_build/sub/tests \
                         TESTSUITEFLAGS="--recheck --verbose"
```